02_cleaning.Rmd

```diff
@@ -1035,3 +1035,212 @@ print(objSDC, "risk")
+  #' find.responses(raw.data, q.db, "responses")
+  
+  # list of choices for each list_name (from TOOL_CHOICES)
+  list.choices <- tool.choices %>% filter(!is.na(list_name)) %>% group_by(list_name) %>%
+    mutate(choices=paste(name, collapse=";\n"),
+           choices.label=paste(!!sym(label_colname), collapse=";\n")) %>%
+    summarise(choices=choices[1], choices.label=choices.label[1])
+  
+  # list of choices for each question
+  select.questions <- tool.survey %>%
+    rename(q.label=label_colname) %>%
+    select(type, name, q.label) %>%
+    mutate(q.type=as.character(lapply(type, function(x) return(str_split(x, " ")[[1]][1]))),
+           list_name=as.character(lapply(type, function(q_type){
+  #' finds the choice list for a question basing on its type
+  q_type.1 <- str_split(q_type, " ")[[1]]
+  if (length(q_type.1)==1) return(NA)
+  else return(q_type.1[2])
+}))) %>%
+    filter(list_name!="NA" & list_name!="group" & list_name!="repeat") %>%
+    left_join(list.choices, by="list_name") %>%
+    filter(!is.na(choices))
+  # create a data frame with the questions selected
+  question.db <- tool.survey %>% filter(name %in% questions) %>%
+    rename(label=label_colname) %>%
+    select("name", "label", "relevant") %>%
+    mutate(ref.name=as.character(lapply(relevant, get.ref.question))) %>%
+    left_join(select(select.questions, "name", "q.type", "q.label", "list_name", "choices", "choices.label"),
+              by=c("ref.name"="name")) %>%
+    rename(ref.label=q.label, ref.type=q.type) %>%
+    mutate(full.label=paste0(ref.label, " - ", label)) %>%
+    select(name, ref.name, full.label, ref.type, choices, choices.label)
+  if(nrow(question.db) == 0){
+      warning("questions is empty - returning an empty dataframe.")
+      return(data.frame())
+  }
+
+  if(nrow(data) == 0){
+      warning("data is empty - returning an empty dataframe.")
+      return(data.frame())
+  }
+
+  responses <- data %>%
+      select(c("uuid", any_of(question.db$name))) %>%
+      pivot_longer(cols = any_of(question.db$name),
+                   names_to="question.name", values_to=values_to,
+                   values_transform = as.character) %>%
+      filter(!is.na(!!sym(values_to))) %>%
+      select(uuid, question.name, !!sym(values_to))
+
+  responses.j <- responses %>%
+      left_join(question.db, by=c("question.name"="name")) %>% dplyr::rename(name="question.name") %>%
+      left_join(select(data, uuid), by="uuid") 
+  return(responses.j)
+}
+```
+
+2. Translate Responses
+
+This function translate a vector from a given dataframe.
+
+It will return the same dataframe as the find.responses function above but with a new column containing the translation. 
+Also, it will return a logframe csv file in the same directory of the running script.
+
+This function will require the remote package `translateREACH`.
+This package can be installed by:
+
+> install.packages(“devtools”)
+> devtools::install_github(“AbrahamAz/translateREACH”)
```

Comment:
Shouldn't  it be installed form "REACH-WoU-Regional" GitHub account, similar to instructions listed in the Translate API SharePoint: 
devtools::install_github(“REACH-WoU-Regional/translateREACH”) ?

Reply:
it should be now REACH-WoU-Regional/translateR. Will also update it on intranet